python/modules/BtagSFWeightQCDProducer.py

```diff
@@ -0,0 +1,144 @@
+import ROOT
+import os
+import numpy as np
+ROOT.PyConfig.IgnoreCommandLineOptions = True
+
+from PhysicsTools.NanoAODTools.postprocessing.framework.datamodel import Collection
+from PhysicsTools.NanoAODTools.postprocessing.framework.eventloop import Module
+
+class BtagSFWeightQCDProducer(Module):
+
+    def __init__(self, bTagEffFile, sampleName, bDiscCut, jetPtMin = 20, jetEtaMax = 2.4, fileDirectory = os.environ['CMSSW_BASE'] + "/src/PhysicsTools/NanoSUSYTools/data/btagSF/"):
+        self.jetPtMin = jetPtMin
+        self.jetEtaMax = jetEtaMax
+        self.bDiscCut = bDiscCut
+
+        self.bTagEffFile = bTagEffFile
+        self.sampleName = sampleName
+        self.fileDirectory = fileDirectory
+
+    def beginJob(self):
+        ROOT.TH1.AddDirectory(False)
+        
+        fin = ROOT.TFile.Open(self.fileDirectory + "/" + self.bTagEffFile)
+
+        self.h_eff_b          = fin.Get(("n_eff_b_" + self.sampleName));
+        self.h_eff_c          = fin.Get(("n_eff_c_" + self.sampleName));
+        self.h_eff_udsg       = fin.Get(("n_eff_udsg_" + self.sampleName));
+        d_eff_b          = fin.Get(("d_eff_b_" + self.sampleName));
+        d_eff_c          = fin.Get(("d_eff_c_" + self.sampleName));
+        d_eff_udsg       = fin.Get(("d_eff_udsg_" + self.sampleName));
+
+        if not self.h_eff_b or not self.h_eff_c or not self.h_eff_udsg:
```

Comment:
Remind me what to expect if fin.Get() failed?

Reply:
If fin.Get fails it will default to using the TTbarInc_2016 values instead of the samplename